File: Data Structures & Algorithms/binary-search/submission-0.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l = 0, r = nums.size() - 1;
        while(l <= r){
           int mid = (l + r)/2;
           if(nums[mid] == target){
            return mid;
           } else if(nums[mid] < target){
            ++l;
            continue;
           } else {
            --r;
            continue;
           }
        }
    
        return -1;
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        output = [1] * len(nums)

        left_multi = 1
        for i in range(len(nums)):
            output[i] = left_multi
            left_multi *= nums[i]

        right_multi = 1
        for i in range(len(nums)-1, -1, -1):
            output[i] *= right_multi
            right_multi *= nums[i]
        
        return output